src/Transports.AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -288,5 +302,39 @@ private async Task<GraphQLRequest> DeserializeFromGraphBodyAsync(Stream bodyStre
 
             return new GraphQLRequest { Query = query }; // application/graphql MediaType supports only query text
         }
+
+#if NET5_0_OR_GREATER
+        private static System.Text.Encoding GetEncoding(HttpRequest request)
+        {
+            if (!request.Headers.TryGetValue("Content-Type", out var value))
+                return null;
+
+            string charset = value.ToString();
+            var position = charset.IndexOf("charset=");
+            if (position == -1)
+                return null;
+
+            charset = charset.Substring(position + 8);
+            if (string.IsNullOrEmpty(charset))
+                return null;
+
+            try
+            {
+                // Remove at most a single set of quotes.
+                if (charset.Length > 2 && charset[0] == '\"' && charset[charset.Length - 1] == '\"')
+                {
+                    return System.Text.Encoding.GetEncoding(charset.Substring(1, charset.Length - 2));
+                }
+                else
+                {
+                    return System.Text.Encoding.GetEncoding(charset);
+                }
```

Comment:
Verify to see if MediaTypeHeaderValue.TryParse removes these quotes automatically...

```suggestion
                return System.Text.Encoding.GetEncoding(charset);
```

Reply:
It does not.

Reply:
Then are you sure it is in the spec?  I would like to see a spec that says it may have quotes before we add code parsing quotes which typically are not there and are not parsed by the .NET code.

Reply:
https://datatracker.ietf.org/doc/html/rfc7231#section-3.1.1.1

https://github.com/dotnet/runtime/issues/15947

https://github.com/dotnet/corefx/pull/33280

Unfortunately, the quote parsing has to stay (for 100% compatibility), and .NET refused to fix their parser because "it might break compatibility".

Reply:
Actually I copied this code from MS sources and then changed a bit.